### PR TITLE
Use static imports and tags

### PR DESCRIPTION
**Pull request by tiraboschi**

### CODE REVIEW

**Comment by tiraboschi** [comment]
Fixes https://github.com/oVirt/ovirt-ansible-hosted-engine-setup/issues/20

File: tasks/main.yml [hunk 1]
```diff
@@ -1,77 +1,19 @@
...
-    - name: iSCSI discover
-      import_tasks: "iscsi_discover.yml"
-      when: he_setup_iscsi_discover
-
-    - name: Get iSCSI devices
-      import_tasks: "iscsi_getdevices.yml"
-      when: he_setup_iscsi_getdevices
+- name: Deploy hosted-engine
+  import_tasks: full_execution.yml
+  tags: [always]
```
Comment (irosenzw) [inline tasks/main.yml]: you don't need [] around the `always` tag

File: tasks/partial_execution.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        - name: 04 Bootstrap local VM
+          import_tasks: bootstrap_local_vm/04_engine_final_tasks.yml
+          delegate_to: "{{ groups.engine[0] }}"
+
+    - name: Add first HE host
+      block:
+        - name: 05 Bootstrap local VM
+          import_tasks: bootstrap_local_vm/05_add_host.yml
+  tags: [bootstrap_local_vm, never]
+
```
Comment (irosenzw) [inline tasks/partial_execution.yml]: redundant line

File: tasks/partial_execution.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+- name: Create hosted-engine storage domain
+  block:
+    - name: Create Storage Domain
+      import_tasks: create_storage_domain.yml
+  tags: [create_storage_domain, never]
+
```
Comment (irosenzw) [inline tasks/partial_execution.yml]: redundant line

File: tasks/partial_execution.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+- name: Create and configure target VM
+  block:
+    - name: Create target hosted engine vm
+      import_tasks: create_target_vm/01_create_target_hosted_engine_vm.yml
+
+    - name: Configure database settings
+      import_tasks: create_target_vm/02_engine_vm_configuration.yml
+      delegate_to: "{{ groups.engine[0] }}"
+  tags: [create_target_vm, never]
+
```
Comment (irosenzw) [inline tasks/partial_execution.yml]: redundant line

File: tasks/partial_execution.yml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+- name: Hosted engine final tasks
+  import_tasks: create_target_vm/03_hosted_engine_final_tasks.yml
+  tags: [create_target_vm, never]
+
```
Comment (irosenzw) [inline tasks/partial_execution.yml]: redundant line

File: tasks/partial_execution.yml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+- name: Final clean
+  import_tasks: final_clean.yml
+  tags: [create_target_vm, final_clean, never]
+
```
Comment (irosenzw) [inline tasks/partial_execution.yml]: redundant line

File: tasks/partial_execution.yml [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+- name: Validate network interface
+  import_tasks: "pre_checks/001_validate_network_interfaces.yml"
+  tags: [get_network_interfaces, never]
+
```
Comment (irosenzw) [inline tasks/partial_execution.yml]: redundant line

File: tasks/partial_execution.yml [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+- name: Validate hostnames
+  import_tasks: "pre_checks/002_validate_hostname_tasks.yml"
+  tags: [validate_hostnames, never]
+
```
Comment (irosenzw) [inline tasks/partial_execution.yml]: redundant line

File: tasks/partial_execution.yml [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+- name: Get FC devices
+  import_tasks: "fc_getdevices.yml"
+  tags: [fc_getdevices, never]
+
```
Comment (irosenzw) [inline tasks/partial_execution.yml]: redundant line

File: tasks/partial_execution.yml [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+- name: iSCSI discover
+  import_tasks: "iscsi_discover.yml"
+  tags: [iscsi_discover, never]
+
```
Comment (irosenzw) [inline tasks/partial_execution.yml]: redundant line

File: tasks/partial_execution.yml [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+      import_tasks: install_packages.yml
+
+    - name: System configuration validations
+      include_tasks: "{{ item }}"
+      with_fileglob: "pre_checks/*.yml"
+
+    - name: Clean environment before deployment
+      import_tasks: initial_clean.yml
+  tags: [initial_clean, bootstrap_local_vm, never]
+
```
Comment (irosenzw) [inline tasks/partial_execution.yml]: redundant line

Comment (tiraboschi) [reply tasks/partial_execution.yml]: I explicitly added double empty lines to keep differently tagged sections more visible

**Comment by tiraboschi** [comment]
Requires oVirt/ovirt-ansible-engine-setup#39

**Comment by tiraboschi** [comment]
Verified together with 
oVirt/ovirt-ansible-engine-setup#39
and
https://gerrit.ovirt.org/#/c/97437/
on
https://jenkins.ovirt.org/view/oVirt%20system%20tests/job/ovirt-system-tests_manual/3999/console

Manually verified for a direct execution

**Pull request merged**